### DISCUSSION TITLE
numpy: cannot import name randbits -- secret namespace collision

### DISCUSSION THREAD

**Discussion by don1uppa** [apache/airflow#29729] [General]
It appears to me that the secrets module is creating a namespace collision with the standard library.

This is causing Pandas to fail with error "numpy: cannot import name randbits"

Here is my code:

    **import secrets
    print(secrets)**

    import pandas as pd
    from imports.support.psycopg_helper import get_connection

    engine = get_connection(conn_name)
    pdf = pd.read_sql(query_sql, con=engine)
    pdf.to_csv(output_path, index=False, header=True)

Here is the output:

**<module 'secrets' from 'C:\\repo\\airflow\\venv\\Lib\\site-packages\\airflow\\secrets\\__init__.py'>**
Traceback (most recent call last):
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "C:\repo\airflow\venv\lib\site-packages\pandas\__init__.py", line 16, in <module>
    raise ImportError(
ImportError: Unable to import required dependencies:
numpy: cannot import name randbits

If I rename the module I get: 
[1:36 PM] Donald Iuppa
**<module 'secrets' from 'C:\\Users\\diuppa\\AppData\\Local\\Programs\\Python\\Python39\\lib\\secrets.py'>**

which is the correct module:

The issue is discussed here:
https://stackoverflow.com/questions/73055157/what-does-importerror-cannot-import-name-randbits-mean

Is it possible to have that module renamed?

**Comment by don1uppa** [comment]
For now I worked around the issue by adding the following code to the end of file: **venv\Lib\site-packages\airflow\secrets\init.py**

import base64

from hmac import compare_digest
from random import SystemRandom

_sysrand = SystemRandom()

randbits = _sysrand.getrandbits
choice = _sysrand.choice

**Reply by potiuk** [reply]
I think it is something wrong with your env. the `secrets` module is part of "airflow.secrets". It would only cause collision if for some reaon you treat "airflow" at PYTHONPATH. This might have something to do that you are trying to use airflow on Windows, which is currently not supported.

**Reply by don1uppa** [reply]
Thank you!   So I figured out base on your answer that I had checked "Add source roots to PYTHONPATH" in my pycharm configuration.  Removing that check resolved the issue.

**Comment by Taragolis** [comment]
> `C:\repo\airflow\venv\Lib\site-packages\airflow\secrets\init.py`

And also please note that Windows not supported/tested, see [Requirements](https://github.com/apache/airflow#requirements)
